### ISSUE TITLE
Daily emails are sent every minute of the hour

### ISSUE THREAD

**Issue by jpohjolainen** [Netflix/security_monkey#939]
## Description of issue:

When having only Daily Email enabled and Change Emails set to None, the daily email are sent every minute of the hour.  So at 12 we start to get each of the 14 auditor report emails every minute.

In the logs I can find this: 2018-02-08 10:26:23,882 INFO: No Audit issues.  Not sending audit email. [in /usr/local/src/security_monkey/security_monkey/auditor.py:815]

We're using latest develop branch (last fetched 7.2.2018) and running on ECS.

**Comment by mikegrima** [comment]
Interesting...

Can you provide the following details:

1. Are the emails for the same exact account?  If so, are they exactly the same?
1. How many scheduler instances do you have running (and how many celery processes. There should only be exactly 1 scheduler running at any time)?
1. Which technologies are you noticing this for (specific, or all)?
1. Was this before or after I merged in this PR: https://github.com/Netflix/security_monkey/pull/935 ?

We are also investigating a bug with Celery about duplicate tasks being scheduled (Celery bug, not SM), and are curious to know if this is related to it.

**Comment by mikegrima** [comment]
https://github.com/celery/celery/issues/4041 may be related.

**Comment by jpohjolainen** [comment]
1. we only have 1 account on SM, so yes, the same account.
2. 1 scheduler, 1 worker with default settings (worker_concurrency=10, worker_max_task_per_child=1)
3. All technologies
4. This is before.

I have now fetched the latest and will deploy new version. I will tell if it fixed this.

**Comment by mikegrima** [comment]
Hello @jpohjolainen 

I just pushed a change that may help address these issues.  The latest develop branch is now the main release candidate for our 1.0 release.  Please update to it and let me know if you are still seeing the behavior in question.

Also, one more question: is this for SES or for SMTP?

**Comment by jpohjolainen** [comment]
Using SMTP. I fetched the changes, but now there hasn't been any daily emails sent. According to logs, there is no activity from 11-12 now.

**Comment by mikegrima** [comment]
Have you confirmed if there were any changes that were made in the environment?

**Comment by jpohjolainen** [comment]
I've pushed same config every time, only the source code has changed inside the image.

**Comment by mikegrima** [comment]
When you say "pushed the same config"... What does this mean?

Is SM properly alerting you on changes made to your AWS environment?

**Comment by jpohjolainen** [comment]
I'm building docker containers and then deploy them to ECS with config in env vars.

Yes, change emails are sent normally.

**Comment by mikegrima** [comment]
I guess I'm confused... Is there actually a problem?

**Comment by jpohjolainen** [comment]
Yes, the daily email dont work. Only change emails, which are sent after auditing works. But we'd like to have once a day reports working like it was earlier.
